### PR TITLE
OKTA-445342 - Enhance device developer docs to include/exclude attribute

### PR DESCRIPTION
**Pull request by apoorvadeshpande-okta**
OKTA-445342 - Enhance device developer docs to include/exclude attribute

Note section -
Screen Shot 2021-11-09 at 5 08 38 PM

## Description:
- **What's changed?**  Enhance device developer docs to include/exclude attribute
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-445342](https://oktainc.atlassian.net/browse/OKTA-445342)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 1]
```diff
@@ -20,6 +20,13 @@ The Devices API supports the following **Authorization Schemes**:
 * SSWS - [API tokens](/docs/reference/core-okta-api/#authentication)
 * Bearer - [OAuth2.0 and OpenID Connect](/docs/concepts/oauth-openid/)
 
+> **Note:** For devices to enroll in Okta and show up in the Devices API, the following actions are required:
+> 1. Admins - Enable Okta FastPass. See [enable FastPass](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/fp/fp-enable.htm)
+> 2. End users with existing mobile Okta Verify enrollments - Migrate existing Okta Verify enrollments to use [FastPass](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/fp/fp-main.htm) after the org migrates to Okta Identity Engine.
+
+> End users with a new enrollment in Okta Verify on an Okta Identity Engine org will have a device record created in the device inventory by default.
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md:27]:
```suggestion
> End users with a new enrollment in Okta Verify on an Okta Identity Engine org have a record created in the device inventory by default.
```

Comment (jeanettedillon-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md:27]: End users with a new enrollment in Okta Verify on an Okta Identity Engine org have a device record created in the device inventory by default.

**Review by jeanettedillon-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md (hunk 1 above)
Comment (apoorvadeshpande-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: @brentschaus-okta I updated the text! Thanks for catching this

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 2]
```diff
@@ -113,18 +118,18 @@ curl -v -X GET \
     "profile": {
         "displayName": "Example Device name",
         "platform": "WINDOWS",
-        "manufacturer": null,
-        "model": null,
-        "osVersion": null,
-        "serialNumber": "Example",
-        "imei": null,
-        "meid": null,
-        "udid": null,
```
Comment (jane-kuang-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md:122]: curious, why are we removing these keys from the json in docs?

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 3]
```diff
@@ -959,11 +964,8 @@ The device model defines several read-only properties:
       "model":"VMware7,1",
       "osVersion":"10.0.18362",
       "serialNumber":"56 4d 4f 95 74 c5 d3 e7-fc 3a 57 9c c2 f8 5d ce",
-      "imei":null,
```
Comment (jane-kuang-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md:962]: Should you be removing `manufacturer`, `model`, `osVersion`, `serialNumber` like in the above cases?

**Review by santoshmale-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 4]
```diff
@@ -113,18 +118,18 @@ curl -v -X GET \
...
         "registered":true,
         "secureHardwarePresent":false
     },
+    "resourceType": "UDDevice",
+    "resourceDisplayName": {
+    "value": "Example Device name",
+      "sensitive": false
+    },
+    "resourceAlternateId": null,
+    "resourceId": "guo1j774nHEZFHsL10w5",
```
Comment (santoshmale-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md:132]: Actually who does `resource..` attributes not show up for `Users` Api? Or at least it is not doc'd

**Comment by santoshmale-okta** [comment]
Screen Shot 2021-11-12 at 4 05 13 PM

Unrelated to this PR, but, The font of a section title `Device Object` seems to be smaller than that of the `Device Properties`. Similarly, `DeviceProfileObject` section.

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md (hunk 4 above)
Comment (apoorvadeshpande-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md:132]: This is an artifact of `Device` object extending `Resource` class.

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 5]
```diff
@@ -20,6 +20,13 @@ The Devices API supports the following **Authorization Schemes**:
 * SSWS - [API tokens](/docs/reference/core-okta-api/#authentication)
 * Bearer - [OAuth2.0 and OpenID Connect](/docs/concepts/oauth-openid/)
 
+> **Note:** For devices to enroll in Okta and show up in the Devices API, the following actions are required:
+> 1. Admins - Enable Okta FastPass. See [enable FastPass](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/fp/fp-enable.htm)
+> 2. End users with existing mobile Okta Verify enrollments - Migrate existing Okta Verify enrollments to use [FastPass](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/fp/fp-main.htm) after the org migrates to Okta Identity Engine.
```
[the remark this answers is not in the document]
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md:25]: @apoorvadeshpande-okta I suggest this change b/c our manager, Maureen, asks that we not use the term "migrate" as part of the OIE upgrades.

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 6]
```diff
@@ -20,6 +20,13 @@ The Devices API supports the following **Authorization Schemes**:
 * SSWS - [API tokens](/docs/reference/core-okta-api/#authentication)
 * Bearer - [OAuth2.0 and OpenID Connect](/docs/concepts/oauth-openid/)
 
+> **Note:** For devices to enroll in Okta and show up in the Devices API, the following actions are required:
+> 1. Admins - Enable Okta FastPass. See [enable FastPass](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/fp/fp-enable.htm)
```
[the remark this answers is not in the document]
Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md:24]: @apoorvadeshpande-okta I don't know why we wouldn't capitalize "Enable" here.

**Review by brentschaus-okta — CHANGES_REQUESTED** [review]

**Review by brentschaus-okta — APPROVED** [review]
LGTM!

**Pull request merged**